### PR TITLE
HDDS-5870. Make Datanode usageinfo command output more readable

### PR DESCRIPTION
**Pull request by siddhantsangwan**
## What changes were proposed in this pull request?

Improved readability of `usageinfo` by showing sizes in appropriate units. Also added the `Total Used` field. Here's a description of the output:

- `Capacity`: Total capacity of the datanode, read from the SCM.
- `Ozone Used`: Space used by Ozone, read from the SCM.
- `Remaining`: Total remaining space, read from the SCM.
- `Total Used`: `Capacity` - `Remaining`, or the total used space.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5870

## How was this patch tested?

Manual testing using docker-compose.
usageinfo

### CODE REVIEW

**Comment by siddhantsangwan** [comment]
@lokeshj1703 please take a look!

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/UsageInfoSubcommand.java [hunk 1]
```diff
@@ -96,19 +97,37 @@ public void execute(ScmClient scmClient) throws IOException {
...
-    percentFormat.setMinimumFractionDigits(5);
+    percentFormat.setMinimumFractionDigits(2);
+    percentFormat.setMaximumFractionDigits(2);
 
     System.out.printf("Usage info for datanode with UUID %s:%n",
         info.getNode().getUuid());
-    System.out.printf("%-10s: %20sB %n", "Capacity", info.getCapacity());
-    System.out.printf("%-10s: %20sB (%s) %n", "SCMUsed", info.getUsed(),
+    // print capacity in a readable format
+    System.out.printf("%-13s: %-21s (%s) %n", "Capacity", capacity + " B",
```
Comment (lokeshj1703) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/UsageInfoSubcommand.java:112]: Let's not print capacity and other data in bytes. GB is sufficient.

**Review by lokeshj1703 — COMMENTED** [review]
@siddhantsangwan The change looks great! I have a minor comment inline.

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/UsageInfoSubcommand.java [hunk 2]
```diff
@@ -96,19 +97,37 @@ public void execute(ScmClient scmClient) throws IOException {
...
+    long used = info.getUsed(), remaining = info.getRemaining();
+    long totalUsed = capacity - remaining;
+    double usedRatio = used / (double) capacity;
+    double remainingRatio = remaining / (double) capacity;
     NumberFormat percentFormat = NumberFormat.getPercentInstance();
-    percentFormat.setMinimumFractionDigits(5);
+    percentFormat.setMinimumFractionDigits(2);
+    percentFormat.setMaximumFractionDigits(2);
 
     System.out.printf("Usage info for datanode with UUID %s:%n",
```
Comment (lokeshj1703) [inline hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/UsageInfoSubcommand.java:109]: Let's also see if it is better to print hostname and ip of the datanode.

File: hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/UsageInfoSubcommand.java (hunk 1 above)
Comment (siddhantsangwan) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/UsageInfoSubcommand.java:112]: Thanks for reviewing. Had a discussion with @mukul1987 regarding this in which we had decided to use both the units(like `hdfs dfsadmin -report`).

Comment (lokeshj1703) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/UsageInfoSubcommand.java:112]: I see. Can we remove the space? I think removing -21 would make it more readable.

Comment (siddhantsangwan) [reply hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/datanode/UsageInfoSubcommand.java:112]: Sure. I'm only doing that to keep the information inside the brackets in one single column.

**Comment by siddhantsangwan** [comment]
Updated look of the command:
usage info

**Review by JacksonYao287 — APPROVED** [review]
thanks @siddhantsangwan for this work! the changes looks good to me

**Review by lokeshj1703 — APPROVED** [review]

**Comment by lokeshj1703** [comment]
@siddhantsangwan Thanks for the contribution! @JacksonYao287 Thanks for the review! I have committed the PR to master branch.

**Pull request merged**